packages/ogmios/src/Provider/TxSubmitProvider/OgmiosTxSubmitProvider.ts

```diff
@@ -82,4 +95,31 @@ export class OgmiosTxSubmitProvider extends RunnableModule implements TxSubmitPr
   async startImpl(): Promise<void> {
     return Promise.resolve();
   }
+
+  private async throwIfHandleResolutionConflict(context: SubmitTxArgs['context']): Promise<void> {
+    if (context?.handles && context.handles.length > 0) {
+      if (!this.#handleProvider) {
+        throw new ProviderError(
+          ProviderFailure.NotImplemented,
+          undefined,
+          'No HandleProvider was set during construction.'
+        );
+      }
+
+      const handleInfoList = await this.#handleProvider.resolveHandles({
+        handles: context.handles.map((handle) => handle.handle)
+      });
+
+      for (const [index, handleInfo] of handleInfoList.entries()) {
+        if (!handleInfo || handleInfo.cardanoAddress !== context.handles[index].resolvedAddresses.cardano) {
```

Comment:
```suggestion
        if (!handleInfo || handleInfo.cardanoAddress !== context.handles[index].cardanoAddress) {
```
?

Reply:
The `context.handles[index].resolvedAddresses.cardano` is correct, because it has the previous data type. 
This is what I tried to say earlier, and the reason I had started the refactor to convert the type to the new type, so that we use `cardano.addresses` instead of `resolvedAddresses` everywhere, because otherwise, it does not work. 447670e499528e136f2ca3888e2e7264b685efd7  
However, the tests that are breaking now, are only the timeout ones that have to do with Ogmios and not with the data types ( thought these should have broken as well as the ts linter ). 
- a refactor to use the new type everywhere and clean up the code accordingly.

Reply:
Edit: 
Ticket: LW-7172
PR: https://github.com/input-output-hk/cardano-js-sdk/pull/780